rust/kernel/thread.rs

```diff
@@ -0,0 +1,390 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A kernel thread (kthread).
+//!
+//! This module allows Rust code to create/wakup/stop a kernel thread.
+
+use crate::c_types;
+use crate::error::{ptr_to_result, Error, KernelResult};
+use crate::{bindings, cstr, CStr};
+
+use alloc::boxed::Box;
+use core::ops::FnOnce;
+
+extern "C" {
+    #[allow(improper_ctypes)]
+    fn rust_helper_get_task_struct(task: *mut bindings::task_struct);
+    #[allow(improper_ctypes)]
+    fn rust_helper_put_task_struct(task: *mut bindings::task_struct);
+}
+
+/// The argument that a thread function can use.
+///
+/// # Todo
+///
+/// Once we make [`crate::file_operations::PointerWrapper`] general, we should
+/// make this as a sub trait of it.
+///
+/// Basically, we just handle object "sending" manually, for a [`Send`] object,
+/// [`ThreadArg::into_pointer`] is used to convert it into a pointer that
+/// kernel's thread creation function will accept. And later in the thread
+/// function, [`ThreadArg::from_pointer`] is used to convert it back.
+pub trait ThreadArg: Send {
+    /// Converts a thread argument to a raw poitner.
+    fn into_pointer(self) -> *const c_types::c_void;
+    /// Converts a raw pointer to a thread argument.
+    ///
+    /// # Safety
+    ///
+    /// This function is marked as unsafe because it should only accept the
+    /// pointer we get from [`ThreadArg::into_pointer()`].
+    unsafe fn from_pointer(ptr: *const c_types::c_void) -> Self;
+}
+
+/// # Todo
+///
+/// Delete this once we make [`crate::file_operations::PointerWrapper`] general.
```

Comment:
```suggestion
/// TODO: delete this once we make [`crate::file_operations::PointerWrapper`] general.
```

Reply:
```rust
/// TODO: Delete ...
```
Right? IOW, the first word after `TODO:` label should be capitalized, similar to `SAFETY:`.? Just try to get the exact rule here ;-)

Reply:
Yeah :-)